oauth2_lib/strawberry.py

```diff
@@ -138,7 +143,7 @@ async def has_permission(self, source: Any, info: OauthInfo, **kwargs) -> bool:
 
 class IsAuthorizedForQuery(BasePermission):
     async def has_permission(self, source: Any, info: OauthInfo, **kwargs) -> bool:  # type: ignore
-        if not oauth2lib_settings.OAUTH2_ACTIVE:
+        if not (oauth2lib_settings.OAUTH2_ACTIVE and oauth2lib_settings.AUTHORIZATION_ACTIVE):
```

Comment:
Or maybe even rewrite this as a decorator, with a return value you can specify and an optional log message?